Change

DecoderRony opened the pull request:
Rename ExperienceDetail interface to Experience and
        move to consts
    Change Experience Component state to accept Experience type data
        inplace of index

DecoderRony committed:
Change
    Rename ExperienceDetail interface to Experience and
        move to consts
    Change Experience Component state to accept Experience type data
        inplace of index

DecoderRony committed:
eliminate ExperienceTimeline component CSS file

HJ1X commented on src/components/Canvas.tsx:
```diff
@@ -8,7 +8,9 @@ const Canvas = () => {
         <ColorModeSwitch />
       </Box>
-      <Outlet />
+      <Box mt={10}>
```
Providing a top margin here means that every page will come with a default top margin. Shouldn't every page have access to the entire canvas and decide its layout internally? 

By doing this we are restricting every page from having access to the top space of canvas. Please reconsider this change.

HJ1X commented on src/components/Experience.tsx:
```diff
@@ -31,28 +26,27 @@ const experienceDetails: ExperienceDetail[] = [
-  const [currentExperienceDataIndex, updateExperienceDataIndex] = useState<
-    undefined | number
-  >(undefined);
+  const [currentExperience, setExperience] = useState<undefined | IExperience>(
```
Again the same convention issue. The convention says that the state updater function should be "setCurrentExperience".

I would suggest installing Sonar lint VS Code extension. It will identify these kind of issues beforehand.

HJ1X commented on src/components/Experience.tsx:
```diff
@@ -31,28 +26,27 @@ const experienceDetails: ExperienceDetail[] = [
-          details={experienceDetails}
-          index={currentExperienceDataIndex}
-        />
+        <ExperienceDetails experienceDetails={currentExperience} />
```
Having the prop name "experienceDetails" for both ExperienceTimeline and ExperienceDetails components where one receives the entire experience list and the other one just the current experience makes interpreting components confusing.

I think we can name the experienceDetails list as just "experiences" because it is a collection of Experience objects as can be derived from its type definition. Then, we can just name the props as "experiences" and "experience" and pass "experiences" and "currentExperience" respectively as the values.

HJ1X commented on src/components/Experience.tsx:
```diff
@@ -31,28 +26,27 @@ const experienceDetails: ExperienceDetail[] = [
         <ExperienceTimeline
           experienceDetails={experienceDetails}
-          updateExperienceDataIndex={updateExperienceDataIndex}
+          setExperience={setExperience}
```
By convention, I think we should name the "setExperience" callback prop as "onExperienceChange".

HJ1X commented on src/components/ExperienceTimeline.tsx:
```diff
@@ -1,23 +1,77 @@
+    "0 0 0 4px grey, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+  let background = "black";
+
+  if (isHovered) {
```
I think we don't need this if block. We can use ternary operators here. To avoid excess cluttering of simple. logic.

HJ1X commented on src/components/ExperienceTimeline.tsx:
```diff
@@ -1,23 +1,77 @@
+  let height = "0.313rem";
+  let top = "0.625rem";
+  let left = "-0.05rem";
+  let boxShadow =
```
We should not directly specify a color name but rather use theme-based global color constants declared in "consts" file.

DecoderRony replied:
Yes. I would have done that in my next commit.

HJ1X commented on src/components/ExperienceTimeline.tsx:
```diff
@@ -1,23 +1,77 @@
-    <Box className={styles.timeline}>
-      {experienceDetails.map((experience, index) => (
+    <Box
+      position="relative"
```
Please check if we can separate these Box components with separate custom components with these styles embedded to avoid cluttering of component logic with component style. If not then, maybe we can define separate objects for these and then spread those here in the Box component

HJ1X commented on src/components/ExperienceTimeline.tsx:
```diff
@@ -1,23 +1,77 @@
-          className={styles.experienceSummary}
-          key={experience.role + index}
-          onClick={() => updateExperienceDataIndex(index)}
+          key={experience.toString()}
```
We should not use entire experience objects' serialized value as key.  We can assign a separate ID to each experience and use it as a key. We can also use the index as a key as I find it difficult to think of a case where we will change the orientation of the experience list once it has been finalized.

DecoderRony committed:
Merge branch 'main' of https://github.com/DecoderRony/portfolio into AD

DecoderRony committed:
remove margin-top

DecoderRony committed:
update styles and add animation in timeline

DecoderRony committed:
Merge branch 'main' of https://github.com/DecoderRony/portfolio into AD

DecoderRony committed:
delete timeline css file

DecoderRony committed:
Timeline component change
    -   split timeline component into smaller components
    -   extract styles from component logic

DecoderRony committed:
add theme color for timeline, add technology field in experience interface

DecoderRony committed:
delete timeline css file

DecoderRony committed:
fix: add key to experienceTimelineContainer component

DecoderRony committed:
Implement useContext in TimelineComponent

when clicking on a timeline content the content should stay zoomed in.
to achieve this we need to update styles on the basis on state. Introduced
useContext to manage and use the state wherever required hence reducing the
tedious task to pass state data by prop drilling through multiple components.

DecoderRony committed:
Implement useContext in TimelineComponent

when clicking on a timeline content the content should stay zoomed in.
to achieve this we need to update styles on the basis on state. Introduced
useContext to manage and use the state wherever required hence reducing the
tedious task to pass state data by prop drilling through multiple components.

DecoderRony committed:
Merge branch 'AD' of https://github.com/DecoderRony/portfolio into AD

DecoderRony committed:
fix error - additional end tag for ContextProvider element

DecoderRony committed:
change ExperienceDetails component prop from "experienceDetails" to "experience"

DecoderRony committed:
refactor timeline component, remove useContext and pass timelineElement component as prop to Timeline Component

HJ1X committed:
Merge branch 'main' into AD
